hedera-node/src/main/java/com/hedera/services/records/TxnAwareRecordsHistorian.java

```diff
@@ -300,4 +307,17 @@ private void save(
 				effPayer, baseRecord, consensusSecond, submittingMember);
 		recordCache.setPostConsensus(txnId, baseRecord.getEnumStatus(), expiringRecord);
 	}
+
+
+	private List<TransactionSidecarRecord> getTimestampedSidecars(
+			final List<TransactionSidecarRecord.Builder> sidecars,
+			final Instant txnTimestamp
+	) {
+		if (isNotEmpty(sidecars)) {
+			return sidecars.stream()
```

Comment:
The overhead of Streams API is too high to use in a `handleTransaction` path. 😞

Reply:
Ah, good to know, fixed as part of [this commit](https://github.com/hashgraph/hedera-services/pull/3625/commits/8f9b60282b5d4090d2c86749bf5597b5e33e2fc2).